pkg/stack/stack.go

```diff
@@ -50,6 +52,24 @@ func Apply(plan types.Plan) error {
 		return githubConfigErr
 	}
 
+	if plan.OAuthProvider == "gitlab" {
+		gitlabConfigErr := generateTemplate("gitlab", plan, types.Gitlab{
+			GitLabInstance:  plan.Gitlab.GitLabInstance,
+			InstallationTag: plan.Gitlab.InstallationTag,
+		})
+		if gitlabConfigErr != nil {
+			return gitlabConfigErr
+		}
+
+	}
+
+	stackErr := generateTemplate("stack", plan, authConfig{
```

Comment:
I wonder whether the selection should be done based on `oauth_provider`. This simplifies the code from extra configuration and is easier for users, but may be confusing if Auth is not enabled.

Reply:
Can you explain?

Reply:
It's about this line: https://github.com/openfaas-incubator/ofc-bootstrap/pull/46/files#diff-b8b32a5985a630c24f4fb2f6abf18d92R61

One may use GitLab without OAuth and then this check would be confusing. 
Do you think it's better to add this instead to the `init.yml`:
```
scm: github
#scm: gitlab
```
?